### PR TITLE
MetaQueryEvaluator refactor

### PR DESCRIPTION
**Pull request by krzkaczor**
* for native routes, slippage is dynamically set to always allow max possible slippage of $100
* make Meta part of SwapParams which fixes a bug where slippage could be overriden but wasn't shown on UI properly,
* replace metaqueryEvaluator with simple function
* extract useSwap function
* tests, tests, tests now useSwap function is properly tested for every supported chain

Flips feature flags to enable airdrop badge and gnosis savings on mainnet.

Closes FRO-665

### CODE REVIEW

File: packages/app/src/features/dialogs/savings/deposit/logic/useDepositIntoSavings.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  formValues: DialogFormNormalizedData
+  marketInfo: MarketInfo
+}
+
+export function useDepositIntoSavings({ formValues, marketInfo }: UseSwapParams): {
+  swapInfo: SwapInfo
+  swapParams: SwapParams
+} {
+  const settings = useActionsSettings()
+  const sdai = marketInfo.findOneTokenBySymbol(TokenSymbol('sDAI'))
```
Comment (yivlad) [inline packages/app/src/features/dialogs/savings/deposit/logic/useDepositIntoSavings.ts:19]:
```suggestion
  const sdai = marketInfo.sDAI
```

File: packages/app/src/config/chain/index.ts [hunk 2]
```diff
@@ -102,6 +106,10 @@ const chainConfig: ChainConfig = {
       import.meta.env.VITE_DEV_GNOSIS_SAVINGS === '1' ? gnosisSavingsInfoQuery : unsupportedSavingsInfoQuery,
     daiSymbol: TokenSymbol('XDAI'),
     sDaiSymbol: TokenSymbol('sDAI'),
+    lifiRoutesWithWaivedFees: [
+      // without PSMs only DAI to sDAI route is has waived fees
+      [TokenSymbol('DAI'), TokenSymbol('sDAI')],
```
Comment (yivlad) [inline packages/app/src/config/chain/index.ts:111]:
```suggestion
      [TokenSymbol('XDAI'), TokenSymbol('sDAI')],
```

File: packages/app/src/domain/exchanges/evaluateSwap.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import BigNumber from 'bignumber.js'
+
+import { LifiWaivedRoutes } from '@/config/chain/types'
+import { SwapMeta, SwapParamsBase } from '@/domain/exchanges/types'
+import { NormalizedUnitNumber, Percentage } from '@/domain/types/NumericValues'
+
+export function evaluateSwap(
+  swap: SwapParamsBase,
+  defaults: Pick<SwapMeta, 'maxSlippage'>,
+  nativeRoutes: LifiWaivedRoutes,
```
Comment (yivlad) [inline packages/app/src/domain/exchanges/evaluateSwap.ts:10]: Is native routes a good name? Is it supposed to represent routes that are likely to go through Maker infrastructure? If so I guess native is not ideal name as for small amounts it more likely gets routed through DEXes

**Review by yivlad — APPROVED** [review]

File: packages/app/src/domain/exchanges/useSwap.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+interface UseSwapArgs {
+  swapParamsBase: SwapParamsBase
+  defaults: { defaultMaxSlippage: Percentage }
+}
+
+export function useSwap({ swapParamsBase, defaults }: UseSwapArgs): {
+  swapInfo: SwapInfo
+  swapParams: SwapParams
+} {
+  const chainId = useOriginChainId()
```
[the remark this answers is not in the document]
Comment (krzkaczor) [reply packages/app/src/domain/exchanges/useSwap.ts:18]: I am keeping it as it is. I created another task to normalize use of these. (i think origin chain id should generally be preferred).

**Pull request merged**